blog post: Auspice 2024 summary (#1089)
nextstrain/nextstrain.org: static-site/content/blog/2024-12-10-auspice-2024.md

- [Full changelog](#full-changelog)

## Focus on Selected

When viewing large phylogenetic trees with a sparse filter, it can be difficult
to really see and understand the selected tips as they occupy a minority of the
vertical space available. Enabling **Focus on Selected** allows the selected
tips to take up more vertical space compared to other tips. For example,
[filtering the SARS-CoV-2 global 6m view to one of the earliest clades, 20I
(Alpha)](https://nextstrain.org/ncov/gisaid/global/6m@2024-12-10?d=tree&f_clade_membership=20I%20%28Alpha%29&p=full):

[![Focus on Selected example](/blog/img/focus-on-selected-example.png)]()

You can toggle this focus on/off in the sidebar:

[![Focus on Selected toggle](/blog/img/focus-on-selected-toggle.png)]()

We [plan](https://github.com/nextstrain/auspice/issues/1368) to continue
exploring better ways to understand sets of sequences in the context of the
larger tree, initially by better use of the horizontal axis which should help
for early clades in the tree such as 20I (Alpha).

## View in Other Platforms

Interoperability between platforms is something we've been big proponents of,

> and are part of [PH4GE](https://pha4ge.org/) to help support this. For many

Reviewer: As per comment in the google doc, can someone double check this

Reviewer 2: Independent of James's comment: during the move from the GDoc to here, this got turned from a parenthetical aside to a secondary clause in the sentence. The sentence flow is very awkward as a result -- the verb tenses are muddled ("have been" vs "are" ) and the "this" reference reaches a long way back for its antecedent. 

I would advocate for changing this to something like "Historically, we have always been strong advocates for interoperability between platforms." and leave it at that, unless there's a specific reason to bring PH4GE into it.

Author: Thanks, I like your suggested sentence. Switched to it:

https://github.com/nextstrain/nextstrain.org/blob/58e0190f82bd5575043b66253378bf2579c87289/static-site/content/blog/2024-12-10-auspice-2024.md#L49-L50

Also, I've split the changes into verbatim GDoc conversion (706eb89d11d033ce2e183bdf6bb91541da6ad91b) and my edits on top of that (ca88c2e9484fd48611ebef3778f889144863eb98) to solicit feedback on changes such as this one.

years, Nextstrain.org has supported [fetching](https://nextstrain.org/fetch)
HTTPS-addressable Auspice JSONs.
[UShER](https://genome.ucsc.edu/cgi-bin/hgPhyloPlace) uses this feature to link
to Auspice views for subtrees.

Other platforms have developed a similar approach to access Auspice datasets,
often using our RESTful API. When viewing datasets in Nextstrain, you can switch
to supported platforms via the new **VIEW IN OTHER PLATFORMS** button at the
bottom of the page:

[![View in Other Platforms button](/blog/img/view-in-other-platforms.png)]()

There are currently three supported platforms:

- [Taxonium](https://taxonium.org) (added in v2.53) – scales well to large
  datasets.

- [MicrobeTrace](https://microbetrace.cdc.gov) by US CDC (added in v2.53)

- [Nextclade](https://clades.nextstrain.org/) (added in v2.54) – allows you to
  add your own sequences (in-browser) to the current tree. Certain requirements
  must be met for this to work as expected, such as the root-sequence being
  defined for the dataset. You may get more reliable results using the manually
  curated reference Nextclade datasets, so please double check results against
  those.

>   [![View in Nextclade preview](/blog/img/view-in-nextclade.png)]()

Reviewer: This image kind of feels like it comes out of nowhere. I think a figure caption explaining how you can go from the cattle-outbreak Nextstrain dataset to Nextclade in order to analyse your data is needed.

Author: How's this as an image?

![View in Nextclade preview](https://github.com/nextstrain/nextstrain.org/blob/85553e7828d8044082a444ee19695638473ec4b4/static-site/public/blog/img/view-in-nextclade.png?raw=true)

Updated in 85553e7828d8044082a444ee19695638473ec4b4

## Performance improvements

As sequencing becomes more routine, the number of samples used in trees has also

> increased. Performance in Auspice tends to slow down for big trees (drawing them

Reviewer 2: ```suggestion
increased. Auspice tends to slow down for large trees (drawing performance 
```

Author: Added as 6cda0a398f54df14206b2175fe447afdbc828429

The Nextstrain team continues to improve the visualisation capabilities of
Auspice, including new features, bug fixes and performance improvements. This
year, we've released 19 versions and wanted to highlight some of the
significant changes.

<!--
number of git tags in 2024 (generated by GitHub Copilot):
git tag --list --format='%(creatordate:short)' | grep '^2024' | wc -l
-->

## Table of contents

- [Focus on Selected](#focus-on-selected)
- [View in Other Platforms](#view-in-other-platforms)
- [Performance improvements](#performance-improvements)
- [Conveying confidence of tips through saturation (version 2.56)](#conveying-confidence-of-tips-through-saturation-version-256)
- [Filtering on all metadata](#filtering-on-all-metadata)

> - [Full changelog](#full-changelog)

Author: This would be nice but `id` tags aren't rendered by marked even though they [should be](https://github.com/markedjs/marked/blob/10020d018d4448b1d2d4612baf650c4e3a5949bf/docs/INDEX.md?plain=1#L13)?

Reviewer 3: `id` is in our [allowed attributes](https://github.com/nextstrain/nextstrain.org/blob/b5af08c677e0053f342ee26f3eba968e6d0b9da1/static-site/app/blog/parseMarkdown.ts#L74), so it should work...

Based on the example you linked to, it seems like it doesn't add id tags automatically and you would have to change the Markdown titles to html

```md
<h2 id="focus-on-selected"> Focus on Selected </h2>
```

Author: Oh right 🤦 some reason I thought that was the rendered version but obviously not.

Looks like it's possible to [extend](https://github.com/markedjs/marked/blob/10020d018d4448b1d2d4612baf650c4e3a5949bf/docs/USING_PRO.md#the-renderer--renderer) the renderer to add an anchor automatically, I'll take a crack at it. If it doesn't work out, I'll use HTML tags instead.

Author: Created a separate PR for this: #1091
